internal/fs/local_file_test.go

```diff
@@ -0,0 +1,212 @@
+package fs_test
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	"github.com/googlecloudplatform/gcsfuse/internal/fs/inode"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsutil"
+	. "github.com/jacobsa/ogletest"
+)
+
+const FileName = "foo"
+const FileContents = "teststring"
+
+type LocalFileTest struct {
+	fsTest
+}
+
+func init() {
+	RegisterTestSuite(&LocalFileTest{})
+}
+
+func (t *LocalFileTest) SetUpTestSuite() {
+	t.serverCfg.ImplicitDirectories = true
+	t.serverCfg.MountConfig = &config.MountConfig{
+		WriteConfig: config.WriteConfig{
+			CreateEmptyFile: false,
+		}}
+	t.fsTest.SetUpTestSuite()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *LocalFileTest) NewFileShouldNotGetSyncedToGCSTillClose() {
+	t.newFileShouldGetSyncedToGCSAtClose(FileName)
+}
+
+func (t *LocalFileTest) NewFileUnderExplicitDirectoryShouldNotGetSyncedToGCSTillClose() {
+	err := os.Mkdir(path.Join(mntDir, "explicit"), dirPerms)
+	AssertEq(nil, err)
+
+	t.newFileShouldGetSyncedToGCSAtClose("explicit/foo")
+}
+
+func (t *LocalFileTest) NewFileUnderImplicitDirectoryShouldNotGetSyncedToGCSTillClose() {
+	AssertEq(
+		nil,
+		t.createObjects(
+			map[string]string{
+				// File
+				"implicitFoo/bar": "",
+			}))
+
+	t.newFileShouldGetSyncedToGCSAtClose("implicitFoo/foo")
+}
+
+func (t *LocalFileTest) newFileShouldGetSyncedToGCSAtClose(fileName string) {
+	// Creating a file shouldn't create file on GCS.
```

Comment:
Shall we extract line 63 - 71 into a separate helper method and use it in other tests?

Reply:
done